spark-sql-application/src/main/scala/org/opensearch/sql/SQLJob.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql
+
+import org.apache.spark.sql.{DataFrame, SparkSession, Row}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.functions._
+
+object SQLJob {
+  def main(args: Array[String]) {
+    // Get the SQL query and Opensearch Config from the command line arguments
+    val query = args(0)
+    val index = args(1)
+    val host = args(2)
+    val port = args(3)
+    val scheme = args(4)
+    val auth = args(5)
+    val region = args(6)
+
+    // Create a SparkSession
+    val spark = SparkSession.builder().appName("SQLJob").getOrCreate()
+
+    try {
+      // Execute SQL query
+      val result: DataFrame = spark.sql(query)
+
+      val resultJson = getJson(result)
+      val schemaJson = getJson(getSchema(spark, result))
+      val data = getList(spark, resultJson, "result").join(getList(spark, schemaJson, "schema")).withColumn("stepId", lit(sys.env.getOrElse("EMR_STEP_ID", "")))
+
+      // Write data to OpenSearch index
+      val aos = Map(
+        "host" -> host,
+        "port" -> port,
+        "scheme" -> scheme,
+        "auth" -> auth,
+        "region" -> region)
+
+      data.write
+        .format("flint")
+        .options(aos)
+        .mode("append")
+        .save(index)
+
+    } finally {
+      // Stop SparkSession
+      spark.stop()
+    }
+  }
+
+  def getJson(df: DataFrame): DataFrame = {
```

Comment:
what is the limitation of the result size? 100MB, because `limits the maximum size of a HTTP request to 100mb`?

Reply:
Yes I think so but I haven't tested with large dataset yet

Reply:
Can we exclude the filed in which we are writing the result from the mapping. 
we can keep the data in _source and the filed shouldn't be analyzed or indexed.